pytorch_lightning/trainer/connectors/checkpoint_connector.py

```diff
@@ -145,6 +146,14 @@ def restore_training_state(self, checkpoint):
         self.trainer.global_step = checkpoint['global_step']
         self.trainer.current_epoch = checkpoint['epoch']
 
+        # crash if max_epochs is lower than the current epoch from the checkpoint
```

Comment:
good catch! this could be a separate PR since it's independent of the checkpoint_on changes

Reply:
agreed, but the tests to verify the new changes were failing because of this bug... let me see if i can put in a different PR and merge first

Reply:
https://github.com/PyTorchLightning/pytorch-lightning/pull/3580/files